
This adds user LED trigger on mmc0.
More useful when system boots from mm0 than dumb blinks of heartbeat....

diff --speed-large-files --no-dereference --minimal -Naur linux-6.2-rc8/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts linux-6.2-rc8/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts
--- linux-6.2-rc8/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2023-02-15 18:57:39.839160833 +0100
+++ linux-6.2-rc8/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2023-02-15 18:50:17.979177021 +0100
@@ -4,6 +4,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/pinctrl/rockchip.h>
+#include <dt-bindings/leds/common.h>
 #include "rk3588.dtsi"
 #include "rk3588s-rk806-dual.dtsi"
 
@@ -12,8 +13,8 @@
 	compatible = "radxa,rock-5b", "rockchip,rk3588";
 
 	aliases {
-		mmc0 = &sdmmc;
-		mmc1 = &sdhci;
+		mmc0 = &sdhci;
+		mmc1 = &sdmmc;
 		serial2 = &uart2;
 	};
 
@@ -39,6 +40,19 @@
 		regulator-max-microvolt = <5000000>;
 		vin-supply = <&vcc12v_dcin>;
 	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_user: led-0 {
+			gpio = <&gpio0 RK_PB7 GPIO_ACTIVE_HIGH>;
+			color = <LED_COLOR_ID_BLUE>;
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc1";
+			pinctrl-names = "default";
+			pinctrl-0 = <&led_user_en>;
+		};
+	};
 };
 
 &sdhci {
@@ -66,9 +80,17 @@
 	cap-sd-highspeed;
 	disable-wp;
 	sd-uhs-sdr104;
-	vmmc-supply = <&vcc_3v3_s3>;
+	vmmc-supply = <&vcc_3v3_s3>;
 	vqmmc-supply = <&vccio_sd_s0>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&sdmmc_bus4 &sdmmc_clk &sdmmc_cmd &sdmmc_det>;
 	status = "okay";
 };
+
+&pinctrl {
+	leds {
+		led_user_en: led_user_en {
+			rockchip,pins = <3 RK_PD5 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.2-rc8/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts linux-6.2-rc8/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
--- linux-6.2-rc8/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2023-02-15 18:57:39.802494167 +0100
+++ linux-6.2-rc8/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2023-02-15 18:50:09.332510676 +0100
@@ -4,6 +4,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/pinctrl/rockchip.h>
+#include <dt-bindings/leds/common.h>
 #include "rk3588s.dtsi"
 #include "rk3588-rk806-single.dtsi"
 
@@ -12,7 +13,8 @@
 	compatible = "radxa,rock-5a", "rockchip,rk3588s";
 
 	aliases {
-		mmc1 = &sdhci;
+		mmc0 = &sdhci;
+		mmc1 = &sdmmc;
 		serial2 = &uart2;
 	};
 
@@ -48,6 +50,19 @@
 		regulator-max-microvolt = <1100000>;
 		vin-supply = <&vcc5v0_sys>;
 	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_user: led-0 {
+			gpios = <&gpio3 RK_PD5 GPIO_ACTIVE_HIGH>;
+			color = <LED_COLOR_ID_BLUE>;
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc1";
+			pinctrl-names = "default";
+			pinctrl-0 = <&led_user_en>;
+		};
+	};
 };
 
 &gmac1 {
@@ -84,6 +99,12 @@
 		};
 
 	};
+
+	leds {
+		led_user_en: led_user_en {
+			rockchip,pins = <3 RK_PD5 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
 };
 
 &sdhci {
